reflect.go

```diff
@@ -72,14 +72,22 @@ func derefPointers(rv reflect.Value) (drv reflect.Value, isPtr bool, isNilPtr bo
 	return
 }
 
-// Returns isVoid=true iff is ultimately nil or empty after (recursive) dereferencing.
-// If isVoid=false, erv is set to the non-nil non-empty valid dereferenced value.
-func isVoid(rv reflect.Value) (erv reflect.Value, isVoid bool) {
+// Returns isDefaultValue=true iff is ultimately nil or empty after (recursive)
+// dereferencing. If isDefaultValue=false, erv is set to the non-nil non-empty
+// non-default dereferenced value.
+// A zero/empty struct is not considered default.
+func isDefaultValue(rv reflect.Value) (erv reflect.Value, isDefaultValue bool) {
 	rv, _, isNilPtr := derefPointers(rv)
 	if isNilPtr {
 		return rv, true
 	} else {
 		switch rv.Kind() {
+		case reflect.Bool:
+			return rv, rv.Bool() == false
+		case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+			return rv, rv.Int() == 0
+		case reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
+			return rv, rv.Uint() == 0
```

Comment:
We should keep an eye on this for amino_rs too /cc @zmanian This might have some minor implications on the kms code for instance.

Reply:
Yeah.  This should be (in later PRs at least) compatible with what Proto3 does.